client/src/components/AddCharacterInfoForm.js

```diff
@@ -0,0 +1,150 @@
+import React, { useState, useEffect } from "react";
+import { withRouter } from "react-router-dom";
+import ErrorList from "./layout/ErrorList";
+import translateServerErrors from "../services/translateServerErrors";
+
+import RatingForm from "./RatingForm";
+import LookForm from "./LookForm";
+
+// take in hunterIndex, characterId, and name as props
+const AddCharacterInfoForm = (props) => {
+  const characterId = props.match.params.charId;
+  const hunterIndex = props.match.params.hunterIndex;
+
+  const [errors, setErrors] = useState([]);
+
+  const [characterData, setCharacterData] = useState({
+    name: "",
+  });
+
+  const getCharacter = async () => {
+    try {
+      const response = await fetch(`/api/v1/characters/${characterId}`);
+      if (!response.ok) {
+        const errorMessage = `${response.status} (${response.statusText})`;
+        const error = new Error(errorMessage);
+        throw error;
+      }
+      const body = await response.json();
+      setCharacterData({
+        ...characterData,
+        name: body.character.name,
+        hunterIndex: body.character.hunterIndex,
+      });
+      return body.character.hunterIndex;
+    } catch (error) {
+      console.error(`Error in fetch: ${error.message}`);
+    }
+  };
+
+  useEffect(() => {
+    getCharacter();
+  }, []);
+
+  const getFormOptions = async (hunterIndex) => {
+    try {
+      const response = await fetch(`/api/v1/hunters/${hunterIndex}`);
+      if (!response.ok) {
+        const errorMessage = `${response.status} (${response.statusText})`;
+        const error = new Error(errorMessage);
+        throw error;
+      }
+      const body = await response.json();
+      setFormOptions({ ...formOptions, look: body.look, ratings: body.ratings });
+    } catch (error) {
+      console.error(`Error in fetch: ${error.message}`);
+    }
+  };
+
+  useEffect(() => {
+    getFormOptions(hunterIndex);
+  }, []);
+
+  const [formOptions, setFormOptions] = useState({
+    look: null,
+    ratings: null,
+  });
```

Comment:
should these default to arrays?

Reply:
I use a guard down at the bottom while I'm waiting for this to be fetched. I prefer to use null in a guard as opposed to an [] because comparing empty arrays is terrifying and it's easier to use !something instead of look.length > 0.